### PR TITLE
feat: add match control flow construct page

### LINKED ISSUE #27: The match Control Flow Construct
- Intro to the match statement
- Patterns that bind to values
- Matching with Option<T>
- Matches are exhaustive
- Catch-all patterns and the _ placeholder

Estimated time: 1 hour

### PR DESCRIPTION
**Pull request by irisdv**
Closes #27 

Time spent: 2 hours

### CODE REVIEW

File: src/ch05-02-the-match-control-flow-construct.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+fn value_in_cents(coin: Coin) -> felt252 {
+    match coin {
+        Coin::Penny(()) => 1,
+        Coin::Nickel(()) => 5,
+        Coin::Dime(()) => 10,
+        Coin::Quarter(())=> 25,
+    }
+}
+```
```
Comment (enitrat) [inline src/ch05-02-the-match-control-flow-construct.md:27]: Missing figure annotation (Listing 6-3)

File: src/ch05-02-the-match-control-flow-construct.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+Let’s break down the `match` in the `value_in_cents` function. First we list the `match` keyword followed by an expression, which in this case is the value `coin`. This seems very similar to a conditional expression used with if, but there’s a big difference: with if, the condition needs to evaluate to a Boolean value, but here it can be any type. The type of coin in this example is the `Coin` enum that we defined on the first line.
+
+Next are the `match` arms. An arm has two parts: a pattern and some code. The first arm here has a pattern that is the value `Coin::Penny` and then the `=>` operator that separates the pattern and the code to run. The code in this case is just the value `1`. Each arm is separated from the next with a comma.
```
Comment (enitrat) [inline src/ch05-02-the-match-control-flow-construct.md:31]: > Coin::Penny

Coin::Penny(()) (I think it works with `Coin::Penny(_)` aswell, I think it looks better. Can you try to see if it works with `Coin::Penny::()` and if it does replace all `(())` by `()`? )

File: src/ch05-02-the-match-control-flow-construct.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        Coin::Dime(()) => 10,
+        Coin::Quarter(state)=> {
+            state.print();
+            25
+        },
+    }
+}
+```
+
+To print the value of a variant of an enum in Cairo, we need to add an implementation for the `print` function:
```
Comment (enitrat) [inline src/ch05-02-the-match-control-flow-construct.md:95]: For the `debug::PrintTrait` trait

File: src/ch05-02-the-match-control-flow-construct.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    fn print(self: UsState) {
+        match self {
+            UsState::Alabama(()) => ('Alabama').print(),
+            UsState::Alaska(()) => ('Alaska').print(),
+        }
+    }
+}
+```
+
+If we were to call `value_in_cents(Coin::Quarter(UsState::Alaska(())))`, `coin` would be `Coin::Quarter(UsState::Alaska)`. When we compare that value with each of the match arms, none of them match until we reach `Coin::Quarter(state)`. At that point, the binding for state will be the value `UsState::Alaska`. We can then use that binding in the printTrait, thus getting the inner state value out of the `Coin` enum variant for `Quarter`.
```
Comment (enitrat) [inline src/ch05-02-the-match-control-flow-construct.md:108]: > binding for state will be the value `UsState::Alaska`

I think UsState::Alaska is the variant, and `UsState::Alaska()` is the value

Comment (enitrat) [reply src/ch05-02-the-match-control-flow-construct.md:108]: > coin would be Coin::Quarter(UsState::Alaska)

coin would be Coin::Quarter(UsState::Alaska())

Comment (enitrat) [reply src/ch05-02-the-match-control-flow-construct.md:108]: > printTrait

PrintTrait

**Review by enitrat — APPROVED** [review]
LGTM! Some small comments to address but it's good to merge afterwards

File: src/ch05-02-the-match-control-flow-construct.md (hunk 2 above)
Comment (irisdv) [reply src/ch05-02-the-match-control-flow-construct.md]: I replaced `Coin::Penny(())` with `Coin::Penny(_)` but I couldn't make it work with `Coin::Penny::()`

**Pull request merged**